tests/test_train.py

```diff
@@ -192,3 +192,10 @@ def test_test_inference_case(dataset_path, inference_file, training_log):
 
     # Clean out the inference folder
     rmtree(inference_file.parent / "test")
+    rmtree(fixture_path / "S3DIS" / "input_0.030")
+    rmtree(fixture_path / "S3DIS" / "original_ply")
+    rmtree(fixture_path / "S3DIS" / "calibration")
+    for p in Path(inference_file.parent).glob("Area4_hallway5.ply_*"):
+        p.unlink()
+    for p in Path(inference_file.parent).glob("Area4_hallway5.ply.*"):
+        p.unlink()
```

Comment:
I think I understand what you're doing there, however this is quite a hack to drop everything during the last unit test (it is OK only if we run it...). These subfolders are created in several of our functional tests, it would be more relevant to drop them each time we created them (but such a strategy would be really expensive in terms of pytest running time, in my humble opinion that's not desirable here).

That's basically a use case for pytest *fixtures*, even if it is tricky to design it.

As a lighter alternative, could you please move these cleaning lines at the beginning of the `conftest.py` module? (I assume that this module is executed each time we run pytest, to be verified.) Then we will drop them **if they exist** at the beginning of each pytest session.

Reply:
I chose to put the deletion inside the fixtures.

Reply:
In my opinion, it does not really modify the first attempt: the fixture (and the unit test) is about inference, while the dropped folders are created after a training process.

Let's say it is OK, we will postpone this point for a further work.

Reply:
Both are made in the same file, so they should be executed both, in the order of the file. As a consequence, I do not see the problem. Furthermore, it was the same principle that was adopted to delete the trained model (the log folder), when you designed the tests.